### DISCUSSION TITLE
How to solve `Cannot convert value to AST: {}` ?

### DISCUSSION THREAD

**Discussion by WinSanPang** [dotansimha/graphql-code-generator#8095] [Q&A]
Hey all,

I'm looking to setup our project to use `graphql-code-generator` (we've previously used Apollo CLI codegen for our other repos but understand it's being deprecated) but running into some issues when running the generator.

**Dependencies (only included relevant ones here but can provide full list):**
```
"dependencies": {
  "@apollo/client": "~3.5.10",
  "@graphql-codegen/cli": "^2.8.1",
  "graphql": "^16.4.0",
  "react": "17.0.1",
  "react-native": "0.64.3",
}
"devDependencies": {
"@apollo/rover": "^0.7.0",
"@graphql-codegen/typescript-operations": "^2.5.1",
"@react-native-community/eslint-config": "^3.0.1",
"@types/react": "~17.0.21",
"@types/react-native": "~0.64.12",
"typescript": "^4.7.4"
}
```

**Scripts (only included relevant ones here but can give full list if needed):**
- `"rover-schema": "npx -p @apollo/rover rover graph introspect http://localhost:3000/graphql_v2 > ./schema.graphql",`
- `"generate": "graphql-codegen"`

**Codegen.yaml file:**
```
schema: './schema.graphql'
documents: './**/**/*.ts'
generates:
  graphql/generated.ts:
    plugins:
      - typescript
      - typescript-operations
```

**Snippet of schema.graphql file:**
```
schema {
  query: Query
  mutation: Mutation
}
"An ISO 8601-encoded date"
scalar ISO8601Date
"An ISO 8601-encoded datetime"
scalar ISO8601DateTime
"Represents untyped JSON"
scalar JSON
"A valid hex code, transported as a string"
scalar ThemeColour
scalar Upload
type AboutInformation {
  description: [String!]!
  recommendation: [String!]
}
type Address implements Node {
  address1: String!
  address2: String
  allowedToChangeToAddress(newAddress: AddressInput!): Boolean!
  city: String!
  country: String!
  deliveryNotes: String
  id: ID!
  postcode: String!
  recipientName: String!
  shippingCountry: ShippingCountry!
}

// ... etc.
```

**TLDR/Context:** 
- We're using React Native + Typescript
- I've ran `rover-schema` to generate our `schema.graphql` file
- I've installed `@graphql-codegen/cli` and installed `@graphql-codegen/typescript-operations` as well as `@graphql-codegen/typescript` as per the Typescript specific guide [here](https://www.graphql-code-generator.com/docs/guides/front-end-typescript-only)
- I've tried to point the `codegen.yaml` file to `'./schema.graphql'`
- When I run `npm run generate`, I get back an error stating `Cannot convert value to AST: {}`

I've had a look at other threads mentioning a similar error but couldn't really find a solution. So any advice or help would be super appreciated 🙏 Lemme know if I should provide any more info/details

**Comment by saihaj** [comment]
can you share a codesandbox where this is breaking?
